sourcery_analytics/metrics/utils.py

```diff
@@ -38,12 +38,30 @@ def method_name(method: astroid.nodes.FunctionDef) -> str:
 @nodedispatch
 @validate_node_type(astroid.nodes.FunctionDef)
 def method_lineno(method: astroid.nodes.FunctionDef) -> int:
+    """Returns the line number of the method.
+
+    Not very useful by itself, but can be combined with other metrics for convenience.
```

Comment:
**issue (complexity):** Consider simplifying the documentation and reevaluating the nested function structure for better readability.

The changes introduced here have increased the complexity of the code, particularly with the verbose inline documentation and the nested function definition within `method_file`. While documentation is valuable, the detailed examples might not be necessary and could clutter the code, making it harder to read. Additionally, nesting functions can complicate understanding the code's flow, especially for newcomers. Consider simplifying the documentation to be concise and evaluating if the nested function could be structured differently to maintain readability and reduce cognitive load. Simplifying these aspects could enhance maintainability and readability without compromising functionality.